### ISSUE TITLE
Crash while running query

### ISSUE THREAD

**Issue by a-monteiro** [databricks/databricks-sql-go#90]
I'm getting the following stack trace occasionally, it seems it sometimes happens on canceling a query. I have no way of replicating it, nor do I know if the databricks driver is the culprit here.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x877b8e]

goroutine 4000 [running]:
github.com/apache/thrift/lib/go/thrift.(*THttpClient).closeResponse(0xc0003fa480)
        /go/pkg/mod/github.com/apache/thrift@v0.17.0/lib/go/thrift/http_client.go:140 +0x8e
github.com/apache/thrift/lib/go/thrift.(*THttpClient).Flush(0xc0003fa480, {0xbe6930?, 0xc000052040})
        /go/pkg/mod/github.com/apache/thrift@v0.17.0/lib/go/thrift/http_client.go:196 +0x4b
github.com/apache/thrift/lib/go/thrift.(*TBinaryProtocol).Flush(0xc000294000?, {0xbe6930?, 0xc000052040?})
        /go/pkg/mod/github.com/apache/thrift@v0.17.0/lib/go/thrift/binary_protocol.go:492 +0x2b
github.com/apache/thrift/lib/go/thrift.(*TStandardClient).Send(0x8?, {0xbe6930, 0xc000052040}, {0xbecb28?, 0xc0001e8230?}, 0x10d678?, {0xb01466?, 0x8?}, {0xbe4488, 0xc000294000})
        /go/pkg/mod/github.com/apache/thrift@v0.17.0/lib/go/thrift/client.go:53 +0x1be
github.com/apache/thrift/lib/go/thrift.(*TStandardClient).Call(0xc0003fa4b0, {0xbe6930, 0xc000052040}, {0xb01466, 0x10}, {0xbe4488?, 0xc000294000?}, {0xbe44b0?, 0xc000294008})
        /go/pkg/mod/github.com/apache/thrift@v0.17.0/lib/go/thrift/client.go:92 +0x91
github.com/databricks/databricks-sql-go/internal/cli_service.(*TCLIServiceClient).ExecuteStatement(0xc00037a4e0, {0xbe6930, 0xc000052040}, 0xc000530000)
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/internal/cli_service/cli_service.go:23735 +0xbe
github.com/databricks/databricks-sql-go/internal/client.(*ThriftServiceClient).ExecuteStatement(0xc000294068, {0xbe69a0, 0xc0004441b0}, 0xbdfe88?)
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/internal/client/client.go:90 +0xb3
github.com/databricks/databricks-sql-go.(*conn).executeStatement(0xc0004be330, {0xbe69a0, 0xc000444180}, {0xc00017c100, 0x532}, {0x0?, 0xc00035e400?, 0xc000444180?})
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/connection.go:280 +0x269
github.com/databricks/databricks-sql-go.(*conn).runQuery(0xc0004be330, {0xbe69a0, 0xc000444180}, {0xc00017c100, 0x532}, {0x1021aa8, 0x0, 0x0})
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/connection.go:167 +0x105
github.com/databricks/databricks-sql-go.(*conn).QueryContext(0xc0004be330, {0xbe6968, 0xc00030b2c0}, {0xc00017c100, 0x532}, {0x1021aa8, 0x0, 0x0})
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/connection.go:142 +0x27f
github.com/databricks/databricks-sql-go.(*stmt).QueryContext(0x7f29af9eba68?, {0xbe6968?, 0xc00030b2c0?}, {0x1021aa8?, 0x0?, 0x0?})
        /go/pkg/mod/github.com/databricks/databricks-sql-go@v0.2.2/statement.go:49 +0x45
database/sql.ctxDriverStmtQuery({0xbe6968, 0xc00030b2c0}, {0xbe6b28, 0xc0001164c8}, {0x1021aa8?, 0x0, 0x0})
        /usr/local/go/src/database/sql/ctxutil.go:82 +0xcc
database/sql.rowsiFromStatement({0xbe6968, 0xc00030b2c0}, {0xbe5e40, 0xc0004be330}, 0xc0002a4340, {0x0, 0x0, 0x0})
        /usr/local/go/src/database/sql/sql.go:2835 +0x17e
database/sql.(*Stmt).QueryContext(0xc0002d66c0, {0xbe6968, 0xc00030b2c0}, {0x0, 0x0, 0x0})
        /usr/local/go/src/database/sql/sql.go:2784 +0x205
github.com/burningalchemist/sql_exporter.(*Query).run(0xc0001326e0, {0xbe6968?, 0xc00030b2c0?}, 0x7f29af9eba68?)
        /go/src/github.com/a-monteiro/sql_exporter/query.go:123 +0xe5
github.com/burningalchemist/sql_exporter.(*Query).Collect(0xc0001326e0, {0xbe6968, 0xc00030b2c0}, 0xc0000b2c60?, 0x0?)
        /go/src/github.com/a-monteiro/sql_exporter/query.go:82 +0x9c
github.com/burningalchemist/sql_exporter.(*collector).Collect.func1(0x0?)
        /go/src/github.com/a-monteiro/sql_exporter/collector.go:87 +0x68
created by github.com/burningalchemist/sql_exporter.(*collector).Collect
        /go/src/github.com/a-monteiro/sql_exporter/collector.go:85 +0x78
```

**Comment by andrefurlan-db** [comment]
Hi @a-monteiro , this is a nasty one. The code is failing to read the response body deep inside thrift code...is the query actually getting canceled?

**Comment by a-monteiro** [comment]
> Hi @a-monteiro , this is a nasty one. The code is failing to read the response body deep inside thrift code...is the query actually getting canceled?

I don't think so. 3 of the submitted queries timed out after 2h. Which I believe is the Databricks timeout for when the client doesn't collect the query results. There are also no canceled queries in the query history.
I had some failed queries (7) before that followed the same pattern
 - Query fails to run (reason unknown)
  - A cancel is issued on this query (seemingly successfully)

```
{"level":"error","connId":"01ed808f-eda7-1a70-a2d1-339f348999e1","corrId":"","queryId":"01ed80a9-6dd9-1e29-9426-0eb1f4f046e8","error":"sentinel context done: context deadline exceeded","time":1671570155,"message":"databricks: failed to run query...
{"level":"debug","time":1671570155,"message":"cancel success"}
```

The crash occurs after one of these failed queries and no cancel message is logged.
This is an extract of the query history of the query that was logged with an error just before the crash.
<img width="470" alt="image" src="https://user-images.githubusercontent.com/36137130/208788890-70aa5f8c-3e1c-46cf-b2ee-35bec9c339ef.png">

**Comment by andrefurlan-db** [comment]
@a-monteiro, which Databricks cloud provider and region? I can try to check what happened on my end.

**Comment by a-monteiro** [comment]
> @a-monteiro, which Databricks cloud provider and region? I can try to check what happened on my end.

AWS, `eu-west-1`

**Comment by a-monteiro** [comment]
Still having reasonably frequent crashes. Did you figure out what happened?

**Comment by andrefurlan-db** [comment]
https://github.com/databricks/databricks-sql-go/pull/92

**Comment by andrefurlan-db** [comment]
Closing this as it should be fixed in `v1.0.1`
